File: Client/packages/logger/config/loggerEnv.ts
/**
 * Logger environment helpers — reads process.env only (no packages/config/env import).
 */

function trimEnv(value: string | undefined): string {
  return (value ?? "").trim();
}

export function isLoggerProduction(): boolean {
  return (process.env.NODE_ENV ?? "development") === "production";
}

export function isLoggerVerboseDev(): boolean {
  return trimEnv(process.env.EXPO_PUBLIC_LOGGER_VERBOSE) === "1";
}

export function parseDevCategoryOverrides(): string[] {
  const raw = trimEnv(process.env.EXPO_PUBLIC_LOGGER_CATEGORIES);
  if (!raw) {
    return [];
  }
  return raw
    .split(",")
    .map((part) => part.trim())
    .filter(Boolean);
}

export function shouldExportLogsToPostHog(): boolean {
  if (isLoggerProduction()) {
    return true;
  }
  return trimEnv(process.env.EXPO_PUBLIC_LOGGER_POSTHOG) === "1";
}

File: Server/logger/config/logger_env.py
"""Logger environment helpers."""

import os


def is_logger_production() -> bool:
    return os.getenv("FLASK_ENV") == "production"


def is_logger_verbose_dev() -> bool:
    return (os.getenv("LOGGER_VERBOSE") or "").strip() == "1"


def parse_dev_category_overrides() -> list[str]:
    raw = (os.getenv("LOGGER_CATEGORIES") or "").strip()
    if not raw:
        return []
    return [part.strip() for part in raw.split(",") if part.strip()]


def should_export_logs_to_posthog() -> bool:
    if is_logger_production():
        return True
    return (os.getenv("LOGGER_POSTHOG") or "").strip() == "1"
